### PR TITLE
accounts/external: handle 0 chainid as not-set for the Clef API

### PR DESCRIPTION
**Pull request by karalabe**

### CODE REVIEW

File: accounts/external/backend.go [hunk 1]
```diff
@@ -218,17 +218,17 @@ func (api *ExternalSigner) SignTx(account accounts.Account, tx *types.Transactio
 		args.MaxFeePerGas = (*hexutil.Big)(tx.GasFeeCap())
 		args.MaxPriorityFeePerGas = (*hexutil.Big)(tx.GasTipCap())
 	default:
-		return nil, fmt.Errorf("Unsupported tx type %d", tx.Type())
+		return nil, fmt.Errorf("unsupported tx type %d", tx.Type())
 	}
 	// We should request the default chain id that we're operating with
 	// (the chain we're executing on)
-	if chainID != nil {
+	if chainID != nil && chainID.BitLen() > 0 {
```
Comment (fjl) [inline accounts/external/backend.go:225]: btw, the canonical zero check we use everywhere is `int.Sign() == 0`.

File: accounts/external/backend.go [hunk 2]
```diff
@@ -218,17 +218,17 @@ func (api *ExternalSigner) SignTx(account accounts.Account, tx *types.Transactio
...
 		args.ChainID = (*hexutil.Big)(chainID)
 	}
 	if tx.Type() != types.LegacyTxType {
 		// However, if the user asked for a particular chain id, then we should
 		// use that instead.
-		if tx.ChainId() != nil {
+		if tx.ChainId() != nil && tx.ChainId().BitLen() > 0 {
```
Comment (fjl) [inline accounts/external/backend.go:231]: Here, `tx.ChainId` cannot be nil.

Comment (karalabe) [reply accounts/external/backend.go]: PTAL

File: accounts/external/backend.go (hunk 1 above)
Comment (karalabe) [reply accounts/external/backend.go]: PTAL

File: accounts/external/backend.go [hunk 3]
```diff
@@ -218,17 +222,17 @@ func (api *ExternalSigner) SignTx(account accounts.Account, tx *types.Transactio
...
 	// (the chain we're executing on)
-	if chainID != nil {
+	if chainID != nil && chainID.Sign() != 0 {
 		args.ChainID = (*hexutil.Big)(chainID)
 	}
 	if tx.Type() != types.LegacyTxType {
 		// However, if the user asked for a particular chain id, then we should
 		// use that instead.
-		if tx.ChainId() != nil {
+		if tx.ChainId().Sign() != 0 {
```
Comment (holiman) [inline accounts/external/backend.go:235]: Don't we need a nil-check here?

Comment (fjl) [reply accounts/external/backend.go:235]: `ChainId()` cannot return nil for non-legacy transactions.

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**